app/models/profile.rb

```diff
@@ -34,7 +21,33 @@ def activate
     end
   end
 
-  def verified?
-    verified_at.present?
+  def plain_pii
+    @_plain_pii ||= Pii::Attributes.new
+  end
+
+  def decrypt_pii(password)
+    Pii::Attributes.new_from_encrypted(encrypted_pii, password)
+  end
+
+  def encrypt_pii(password, pii = plain_pii)
+    ssn = pii.ssn
+    encryptor = Pii::Encryptor.new
+    self.ssn_signature = Digest::SHA256.hexdigest(encryptor.sign(ssn)) if ssn
```

Comment:
So, `encryptor.sign()` generates a new key on each call, so as far as I can tell, calling this twice in a row does not result in catching duplicates. Did I read this wrong?